File: Python/551. Student Attendance Record I.py
class Solution:
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        f1, f2 = 0, 0
        for c in s:
            if c == 'A':
                f1 += 1
            if c == 'L':
                f2 += 1
            else:
                f2 = 0
            if f1 > 1 or f2 > 2:
                return False
        return True

File: C++/551. Student Attendance Record I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int c1 = 0, c2 = 0;
        for(auto c: s)
        {
            if(c=='A') c1++;
            if(c=='L') c2++;
            else c2 = 0;
            if(c1>1 || c2>2)
                return false;
        }
        return true;
    }
};